File: src/lib/prompts.ts
export type PromptInputType = "text" | "textarea" | "slider" | "toggle" | "choices";

export interface PromptDefinition {
  id: string;
  label: string;
  category: string;
  inputType: PromptInputType;
  placeholder?: string;
  sliderMin?: number;
  sliderMax?: number;
  toggleLabels?: [string, string];
  choices?: string[];
  privateByDefault?: boolean;
}

export const PROMPT_CATEGORIES = [
  "Recap",
  "Story",
  "Social",
  "Party",
  "Chaotic",
  "Reflection",
  "The plan",
] as const;

export const PROMPTS: PromptDefinition[] = [
  {
    id: "chaos",
    label: "Chaos level",
    category: "Recap",
    inputType: "slider",
    sliderMin: 1,
    sliderMax: 10,
  },
  {
    id: "whoWasDrunkest",
    label: "Who was most gone",
    category: "Recap",
    inputType: "text",
    placeholder: "Name",
  },
  {
    id: "funniestThing",
    label: "The funniest bit",
    category: "Recap",
    inputType: "textarea",
    placeholder: "The one you'll all repeat tomorrow",
  },
  {
    id: "quoteOfNight",
    label: "Line of the night",
    category: "Recap",
    inputType: "text",
    placeholder: "What someone actually said",
  },
  {
    id: "kissedAnyone",
    label: "Anyone get kissed?",
    category: "Recap",
    inputType: "toggle",
    toggleLabels: ["Yes", "No"],
    privateByDefault: true,
  },
  {
    id: "kissedWho",
    label: "Who with?",
    category: "Recap",
    inputType: "text",
    placeholder: "Optional",
  },
  {
    id: "homeTime",
    label: "Home time",
    category: "Recap",
    inputType: "text",
    placeholder: "3am, sunrise, no idea",
  },
  {
    id: "mainCharacter",
    label: "Who ran it",
    category: "Story",
    inputType: "textarea",
    placeholder: "Who thought the night was about them",
  },
  {
    id: "plotTwist",
    label: "The turn",
    category: "Story",
    inputType: "textarea",
    placeholder: "When it stopped going to plan",
  },
  {
    id: "mostEmbarrassing",
    label: "The cringe",
    category: "Story",
    inputType: "textarea",
    placeholder: "The bit you'd rather skip",
  },
  {
    id: "highlight",
    label: "Best bit",
    category: "Story",
    inputType: "textarea",
  },
  {
    id: "lowestPoint",
    label: "When it dipped",
    category: "Story",
    inputType: "textarea",
  },
  {
    id: "nightMvp",
    label: "Who carried",
    category: "Social",
    inputType: "text",
  },
  {
    id: "lostSoldier",
    label: "Who we lost",
    category: "Social",
    inputType: "text",
    placeholder: "Last seen…",
  },
  {
    id: "unexpectedLegend",
    label: "Who came out of nowhere",
    category: "Social",
    inputType: "text",
  },
  {
    id: "metAnyoneNew",
    label: "Anyone new?",
    category: "Social",
    inputType: "text",
  },
  {
    id: "songOfNight",
    label: "The song",
    category: "Party",
    inputType: "text",
    placeholder: "What was playing",
  },
  {
    id: "drinkOfChoice",
    label: "What you were drinking",
    category: "Party",
    inputType: "text",
  },
  {
    id: "danceFloorRating",
    label: "Dancefloor",
    category: "Party",
    inputType: "slider",
    sliderMin: 1,
    sliderMax: 10,
  },
  {
    id: "bestLocation",
    label: "Best stop",
    category: "Party",
    inputType: "text",
  },
  {
    id: "biggestRedFlag",
    label: "The warning",
    category: "Chaotic",
    inputType: "textarea",
    placeholder: "The bit that should've been a no",
  },
  {
    id: "memoryGap",
    label: "Missing hours",
    category: "Chaotic",
    inputType: "textarea",
  },
  {
    id: "morningInvestigation",
    label: "What you reconstructed",
    category: "Chaotic",
    inputType: "textarea",
    placeholder: "Receipts, photos, group chat",
  },
  {
    id: "regretLevel",
    label: "Regret",
    category: "Chaotic",
    inputType: "slider",
    sliderMin: 1,
    sliderMax: 10,
  },
  {
    id: "coreMemory",
    label: "The bit you'll keep",
    category: "Reflection",
    inputType: "textarea",
  },
  {
    id: "oneWordVibe",
    label: "One word",
    category: "Reflection",
    inputType: "text",
    placeholder: "One word. That's it.",
  },
  {
    id: "wouldRepeat",
    label: "Do it again?",
    category: "Reflection",
    inputType: "toggle",
    toggleLabels: ["Yes", "No"],
  },
  {
    id: "tonightsObjective",
    label: "The plan",
    category: "The plan",
    inputType: "text",
    placeholder: "What you were going for",
  },
  {
    id: "missionResult",
    label: "Did it land?",
    category: "The plan",
    inputType: "text",
  },
  {
    id: "bonusAchievement",
    label: "Unplanned win",
    category: "The plan",
    inputType: "text",
  },
  {
    id: "generalComment",
    label: "Anything else",
    category: "Reflection",
    inputType: "textarea",
    placeholder: "The rest of it",
  },
];

export const DEFAULT_PROMPT_IDS = [
  "chaos",
  "whoWasDrunkest",
  "funniestThing",
  "kissedAnyone",
  "songOfNight",
  "oneWordVibe",
  "tonightsObjective",
  "generalComment",
];

export const TIMELINE_EMOJIS = ["🥂", "🏠", "🎉", "🍻", "🎶", "🕺", "🏃", "📱", "🍕", "☕", "🌟", "💃", "🎭", "🔥", "💫"];

File: flutter_app/lib/data/prompts.dart
enum PromptInputType { text, textarea, slider, toggle, choices }

class PromptDefinition {
  const PromptDefinition({
    required this.id,
    required this.label,
    required this.category,
    required this.inputType,
    this.placeholder,
    this.sliderMin = 1,
    this.sliderMax = 10,
    this.toggleLabels = const ['Yes', 'No'],
    this.choices = const [],
    this.privateByDefault = false,
    this.canBePrivate = false,
  });

  final String id;
  final String label;
  final String category;
  final PromptInputType inputType;
  final String? placeholder;
  final int sliderMin;
  final int sliderMax;
  final List<String> toggleLabels;
  final List<String> choices;
  final bool privateByDefault;
  final bool canBePrivate;
}

const hiddenPromptIds = {'chaos'};

const defaultPromptIds = [
  'whoWasDrunkest',
  'funniestThing',
  'songOfNight',
  'oneWordVibe',
  'generalComment',
];

const promptDefinitions = [
  PromptDefinition(
    id: 'whoWasDrunkest',
    label: 'Who was the drunkest',
    category: 'Recap',
    inputType: PromptInputType.text,
    placeholder: 'Name',
    canBePrivate: true,
  ),
  PromptDefinition(
    id: 'funniestThing',
    label: 'The funniest bit',
    category: 'Recap',
    inputType: PromptInputType.textarea,
    placeholder: "The one you'll all repeat tomorrow",
    canBePrivate: true,
  ),
  PromptDefinition(
    id: 'quoteOfNight',
    label: 'Line of the night',
    category: 'Recap',
    inputType: PromptInputType.text,
    placeholder: 'What someone actually said',
  ),
  PromptDefinition(
    id: 'kissedAnyone',
    label: 'Anyone get kissed?',
    category: 'Recap',
    inputType: PromptInputType.toggle,
    privateByDefault: true,
  ),
  PromptDefinition(
    id: 'kissedWho',
    label: 'Who with?',
    category: 'Recap',
    inputType: PromptInputType.text,
    placeholder: 'Optional',
    privateByDefault: true,
  ),
  PromptDefinition(
    id: 'homeTime',
    label: 'Home time',
    category: 'Recap',
    inputType: PromptInputType.text,
    placeholder: '3am, sunrise, no idea',
  ),
  PromptDefinition(
    id: 'songOfNight',
    label: 'The song',
    category: 'Party',
    inputType: PromptInputType.text,
    placeholder: 'What was playing',
  ),
  PromptDefinition(
    id: 'oneWordVibe',
    label: 'One word',
    category: 'Reflection',
    inputType: PromptInputType.text,
    placeholder: "One word. That's it.",
  ),
  PromptDefinition(
    id: 'tonightsObjective',
    label: 'Mission of the night',
    category: 'Mission',
    inputType: PromptInputType.text,
    placeholder: 'What was the mission?',
  ),
  PromptDefinition(
    id: 'missionResult',
    label: 'Mission result',
    category: 'Mission',
    inputType: PromptInputType.text,
  ),
  PromptDefinition(
    id: 'generalComment',
    label: 'Notable mentions',
    category: 'Reflection',
    inputType: PromptInputType.textarea,
    placeholder: 'Anyone or anything else worth logging',
    canBePrivate: true,
  ),
  PromptDefinition(
    id: 'nightMvp',
    label: 'Who carried',
    category: 'Social',
    inputType: PromptInputType.text,
  ),
  PromptDefinition(
    id: 'drinkOfChoice',
    label: 'What you were drinking',
    category: 'Party',
    inputType: PromptInputType.text,
  ),
  PromptDefinition(
    id: 'coreMemory',
    label: "The bit you'll keep",
    category: 'Reflection',
    inputType: PromptInputType.textarea,
  ),
];

final promptLabelById = {
  for (final prompt in promptDefinitions) prompt.id: prompt.label,
};

List<PromptDefinition> defaultPrompts() {
  final byId = {for (final p in promptDefinitions) p.id: p};
  return [
    for (final id in defaultPromptIds)
      if (byId[id] != null) byId[id]!,
  ];
}

List<PromptDefinition> extraPrompts() {
  final defaults = defaultPromptIds.toSet();
  return [
    for (final prompt in promptDefinitions)
      if (!defaults.contains(prompt.id) && !hiddenPromptIds.contains(prompt.id))
        prompt,
  ];
}

String promptDisplayLabel(String id) => promptLabelById[id] ?? id;
